shell/src/main/java/tachyon/shell/TfsShell.java

```diff
@@ -174,6 +178,61 @@ public void load(TachyonURI filePath) throws IOException {
   }
 
   /**
+   * Persist a file or directory currently stored only in Tachyon to the UnderFileSystem
+   *
+   * @param filePath the TachyonURI path to persist to the UnderFileSystem
+   * @throws IOException
+   */
+  public void persist(TachyonURI filePath) throws IOException {
+    try {
+      TachyonFile fd = mTfs.open(filePath);
+      FileInfo fInfo = mTfs.getInfo(fd);
+      if (fInfo.isFolder) {
+        List<FileInfo> files = mTfs.listStatus(fd);
+        List<String> errorMessages = new ArrayList<String>();
+        for (FileInfo file : files) {
+          TachyonURI newPath = new TachyonURI(file.getPath());
+          try {
+            persist(newPath);
```

Comment:
Can't `newPath` be a directory if we have multiple levels of nesting? It looks like `listStatus` only returns direct children of the path, right?